### PR TITLE
MGMT-15345: Adds pkg/release unit-tests

### PR DESCRIPTION
**Pull request by nmagnezi**

### CODE REVIEW

**Comment by nmagnezi** [comment]
tested okay:
```
INFO Reusing base CoreOS ISO from cache
INFO Reusing recovery CoreOS ISO from cache
INFO Reusing data ISO from cache
INFO Successfully downloaded appliance base disk image
INFO Successfully extracted appliance base disk image
INFO Successfully generated appliance disk image
INFO Time elapsed: 3m16s
INFO
INFO Appliance disk image was successfully created in assets directory: assets/appliance.raw
```

/assign @danielerez 
/assign @jhernand

File: pkg/release/release.go [hunk 1]
```diff
@@ -36,6 +36,46 @@ const (
 	ocAdmReleaseInfo     = "oc adm release info quay.io/openshift-release-dev/ocp-release:%s-%s -o json"
 )
 
+type OSInterface interface {
```
Comment (danielerez) [inline pkg/release/release.go:39]: Nice:)
But just for reusability, maybe worth extracting to another pkg (e.g. pkg/fileutil). So we could use it nicely on other places.
Also, about the naming, maybe something like this instead: https://go.dev/talks/2012/10things.slide#8

Comment (nmagnezi) [reply pkg/release/release.go:39]: we have several packages involved in this. I think I can do that but in another PR since it will alter several pkgs.

Comment (danielerez) [reply pkg/release/release.go:39]: ok, so maybe just rename it. The naming here looks nice https://go.dev/talks/2012/10things.slide#8

Comment (nmagnezi) [reply pkg/release/release.go:39]: so just `osFS`?

Comment (nmagnezi) [reply pkg/release/release.go:39]: done

Comment (nmagnezi) [reply pkg/release/release.go]: @danielerez: Added `OSInterface` in https://github.com/openshift/appliance/pull/120
This PR is rebased on top of it.

**Review by danielerez — COMMENTED** [review]
(partially review)

Any change to extract the cincinnati metadata to a separate json file? My browser can't seem to handle it:)

File: pkg/release/release.go (hunk 1 above)
Comment (nmagnezi) [reply pkg/release/release.go]: now rebased on master.

**Comment by nmagnezi** [comment]
> (partially review)
> 
> Any change to extract the cincinnati metadata to a separate json file? My browser can't seem to handle it:)

Sure. Done.

**Pull request merged**